mitodl/mit-open
# Improve channel API performance

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4886

### Description (What does it do?)
- Changes the channels API pagination to default (10/100) instead of large (1000/1000)
- Adds an index on `Channel.channel_type`


### How can this be tested?
- Set DEBUG=False in your backend .env file (reponse time can be >2x slower when True)
- In postman or equivalent, send multiple GET requests to the following, the majority should be speedy, ~250 ms or better (postman will show the response time and you can also check the performance with [silk](http://api.open.odl.local:8063/silk/requests/))
  - http://api.open.odl.local:8063/api/v0/channels/
  - http://api.open.odl.local:8063/api/v0/channels/unit/ocw/
  - http://api.open.odl.local:8063/api/v0/channels/unit/mitx/

### Additional Context
The `/api/v1/channels/type/unit/*` response times referenced in the ticket issue were acceptably fast locally before and after these changes.  But `/api/v1/channels/` was originally quite slow because it was returning all channels in a single response.
